src/orca/services/synapse/client_factory.py

```diff
@@ -0,0 +1,70 @@
+from pydantic.dataclasses import dataclass
+from synapseclient import Synapse
+from synapseclient.core.exceptions import SynapseAuthenticationError
+
+from orca.errors import ConfigError
+from orca.services.base.client_factory import BaseClientFactory
+from orca.services.synapse.config import SynapseConfig
+
+
+@dataclass(kw_only=False)
+class SynapseClientFactory(BaseClientFactory):
+    """Factory for constructing Synapse clients.
+
+    Attributes:
+        config: Configuration object for this service.
+
+    Class Variables:
+        client_class: The client class for this service.
+    """
+
+    config: SynapseConfig
+
+    client_class = Synapse
+
+    def create_client(self) -> Synapse:
+        """Create an authenticated client.
+
+        Typically, this involves pulling values from the configuration,
+        ensuring that non-optional arguments are not set to None, and
+        using them to instantiate a client class.
+
+        Raises:
+            ConfigError: If the configuration is invalid.
+
+        Returns:
+            An authenticated client object.
+        """
+        client = Synapse()
+        auth_token = self.config.auth_token
+
+        if auth_token is None:
+            message = f"Config ({self.config}) is missing auth_token."
+            raise ConfigError(message)
+
+        # Ignore authentication error (leave that to `test_client_request`)
+        try:
+            client.login(authToken=auth_token)
+        except SynapseAuthenticationError:
```

Comment:
Can we check if there are other types of authentication errors thrown by the client?

I'm forgetting if all auth errors extend from this error class.